File: 3. Functions And Array/rotate.py
a = [1, 2, 3, 4, 5]
k = 3
def reverse(a, start, end):
    while(start < end):
        temp = a[end]
        a[end] = a[start]
        a[start] = temp
        start+=1
        end-=1
    return a

def rotate(a, k):
    
    if(k<0):
        k = k + len(a)
    k = k%len(a)
    #part1
    s1 = 0
    e1 = len(a) - k - 1
    reverse(a, s1, e1)
    #part2
    s2 = len(a) - k
    e2 = len(a) - 1
    reverse(a, s2, e2)
    #all
    s = 0
    e = len(a) - 1
    ans = reverse(a, s, e)
    print(ans)

rotate(a, k)

File: 1. getting started/rotate.java
public class rotate {
    public static void main(String[] args) {
        int n = 562984;
        int k = 2;
        int temp = n;
        int nod = 0;

        while(temp!=0){
            temp = temp/10;
            nod++;
        }

        k = k % nod;
        if(k<0){
            k = k + nod;
        }
        int d = 1;
        int m = 1;
        for(int i=1; i<=nod; i++){
            if(i<=k){
                d = d*10;
            }
            else{
                m = m*10;
            }
        }
        int q = n / d;
        int r = n % d;

        int res = r*m + q;
        System.out.println(res);
    }
}
